### PR TITLE
Integer conversion fixes: Allow boolean -> integer

### LINKED ISSUE #68: mutate convert to int converts no data to 0
when we are converting some values to integer, if value is null then we see its value as 0, which gives wrong information in case where for some values we get values as 0.

EX:
if message is SignalStrength=
then we see #SignalStrength 0 as KV pair
we expect blank vlaue.

since in some message we get SignalStrength=0
then we cannot differentiate between if the value is real 0 or it is created by logtsatsh as 0 when there is no data in value.

### LINKED ISSUE #107: Allow boolean -> integer conversions
The code support integer -> boolean conversion (1 = true, 0 = false). We should support the other direction boolean -> integer as well.

### PR DESCRIPTION
**Pull request by jakelandis**
Fixes #107 - Allow boolean -> integer conversions

### CODE REVIEW

File: spec/filters/mutate_spec.rb [hunk 1]
```diff
@@ -537,6 +537,53 @@ def pattern_path(path)
     end
   end
 
+  describe "convert booleans to integer" do
+    config <<-CONFIG
+      filter {
+        mutate {
+          convert => [ "[foo][0]", "integer" ]
```
Comment (yaauie) [inline spec/filters/mutate_spec.rb]: I believe that the hash syntax is preferred:

~~~
filter {
  mutate {
    convert => {"[foo][0]" => "integer"}
    convert => {"[foo][1]" => "integer"}
    convert => {"[foo][2]" => "integer"}
    convert => {"[foo][3]" => "integer"}
    convert => {"[foo][4]" => "integer"}
  }
}
~~~

OR:

~~~
mutate {
  convert => {
    "[foo][0]" => "integer"
    "[foo][1]" => "integer"
    "[foo][2]" => "integer"
    "[foo][3]" => "integer"
    "[foo][4]" => "integer"
  }
}
~~~

File: lib/logstash/filters/mutate.rb [hunk 2]
```diff
@@ -285,7 +289,10 @@ def convert_string(value)
   end
 
   def convert_integer(value)
-    value.to_i
+    true.eql?(value) ? 1  #return 1 if a true boolean
```
Comment (yaauie) [inline lib/logstash/filters/mutate.rb]: the nested ternary is pretty hard to follow; the following is equivalent, and requires fewer comments because the code is less opaque:

~~~ ruby
def convert_integer(value)
  return 1 if value == true
  return 0 if value == false

  return value if value.nil?
  return value if value.kind_of?(String) && value.empty?

  value.to_i
end
~~~

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 3.2.0
+  - Support boolean to integer conversion #107
+  - Don't convert empty/null values to zero for integer conversions #68
```
Comment (yaauie) [inline CHANGELOG.md]: Should we leave it an unconverted string, or should we convert an empty string to nil? I think the latter would produce less surprising results, especially considering the following scenario:

 - field "foo" on event A is populated with a string representing an integer
 - field "foo" on event B is populated with an empty string
 - field "foo" is converted to an integer (`mutate { convert { foo => "integer" }}`)

 - elasticsearch requires consistent types, but will attempt to coerce at index time
   - if the first event for an index was A, the index will be created with foo's type as an integer
   - if the first event for an index was B, the index will be created with foo's type as a string.

Additionally, should the whitespace-only string `"    "` be considered empty for this purpose, or only `""`?

File: spec/filters/mutate_spec.rb (hunk 1 above)
Comment (jakelandis) [reply spec/filters/mutate_spec.rb]: You are right, the latter should be preferred since the former implies that you can slip in some other mutate action between converts such that the other mutate action would trigger between converts converts.... but it doesn't actually work that way (it loops over all the converts (or any single action) all at once.

File: lib/logstash/filters/mutate.rb (hunk 2 above)
Comment (jakelandis) [reply lib/logstash/filters/mutate.rb]: ....but I ❤️ ? :  ? : ? : ? 's 

Will update as suggested 😄

File: CHANGELOG.md (hunk 3 above)
Comment (jakelandis) [reply CHANGELOG.md]: I agree with you that this has the potential to increase mapping errors.  

I will pull this from this PR and discuss further on #68 , it seems that the reporter really wants a way to differentiate a real zero from an invalid converted one.

**Comment by jakelandis** [comment]
@yaauie - mind taking another look ?  I nuked the empyt/nil change and only focused on the boolean->integer conversion.

File: docs/index.asciidoc [hunk 4]
```diff
@@ -58,7 +58,7 @@ filter plugins.
 
 Convert a field's value to a different type, like turning a string to an
 integer. If the field value is an array, all members will be converted.
-If the field is a hash, no action will be taken.
+If the field is a hash, nil, or an empty string, no action will be taken.
```
Comment (yaauie) [inline docs/index.asciidoc]: comment refers to functionality that was rolled back

**Comment by yaauie** [comment]
aside from the doc comment above, LGTM.

File: docs/index.asciidoc (hunk 4 above)
Comment (jakelandis) [reply docs/index.asciidoc]: doh! ... change incoming.

**Comment by jakelandis** [comment]
@yaauie - thanks!  (doc fixed)
